source/MaterialXRuntime/RtFileIo.h

```diff
@@ -30,9 +30,15 @@ class RtReadOptions
     RtReadOptions();
     ~RtReadOptions() { }
 
-    /// If true, duplicate elements with non-identical content will be skipped;
-    /// otherwise they will trigger an exception.  Defaults to false.
-    bool skipConflictingElements;
+    /// Conflict resolution enumeration
+    enum ConflictResolution {
+        SKIP_ELEMENTS, /// Skip conflicting elements
+        RENAME_ELEMENTS, /// Rename conflicting elements
+        THROW_ERROR /// Throw error and leave destination unmodified
+    };
```

Comment:
Out of curiosity, what are the use-cases for having this controllable by an option?

To my mind there are two separate cases _definitions_ vs. _content_. Where _definitions_ (nodedefs, typedefs, attributedefs, implementations, etc.) could never be renamed and duplicates must always be skipped. Whereas for _content_ (nodes, nodegraphs, looks, etc.) we could always resolve the conflict by renaming.

Reply:
Indeed an option that is never used will accumulate bugs. Are you OK with making "rename content, skip definitions" the default behavior and to remove this enum?

Reply:
For me that would be fine if I'm not missing some use-case.. @bernardkwok?